### PR TITLE
feat: [HL-16] Add fonts and provide access for it  🎨

### PR DESCRIPTION
**Pull request by abdallah-7698**
## What type of PR is this? (check all applicable)

- [x] 🎨 Style

## Added tests?

- [x] 🙅 no, because they aren't needed

## Added to documentation?

- [x] 🙅 no documentation needed

### CODE REVIEW

**Review by ahmdmhasn — CHANGES_REQUESTED** [review]
Please drop any commits that aren't relevant to the pull request. 

Screenshot 2023-06-01 at 11 19 30 PM

File: Healthy/Classes/Extensions/UIFont+Style.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+extension UIFont {
+    static let splashHeaderLabel    = appFont(ofSize: 18, weight: .semibold)
+    static let splashTitle          = appFont(ofSize: 50, weight: .semibold)
+    static let splashSubtitle       = appFont(ofSize: 16, weight: .regular)
+    static let signinTitle          = appFont(ofSize: 30, weight: .semibold)
+    static let signinSubtitle       = appFont(ofSize: 20, weight: .regular)
+    static let signupTitle          = appFont(ofSize: 20, weight: .semibold)
+    static let signupSubtitle       = appFont(ofSize: 11, weight: .regular)
+    static let textFieldTitleLabel  = appFont(ofSize: 14, weight: .regular)
+}
```
Comment (ahmdmhasn) [inline Healthy/Classes/Extensions/UIFont+Style.swift]: Please avoing naming fonts with modules.

File: Healthy/Classes/Extensions/UIFont+Style.swift [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    }
+}
+
+// MARK: - Buttons
+
+extension UIFont {
+    static var button           = appFont(ofSize: 16, weight: .semibold)
+    static var smallButton      = appFont(ofSize: 11, weight: .semibold)
+    static var plainGoldButton  = appFont(ofSize: 11, weight: .regular)
+}
```
Comment (ahmdmhasn) [inline Healthy/Classes/Extensions/UIFont+Style.swift]: Same comment as below.

File: Healthy/Classes/Extensions/UIFont+Style.swift [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import UIKit
```
Comment (ahmdmhasn) [inline Healthy/Classes/Extensions/UIFont+Style.swift:1]: Please use the same fonts provided in Figma with exactly the same names. 

https://www.figma.com/file/8ilruINbV0EBVQZP2IQnwB/Food-Recipe-App-(Community)?type=design&node-id=5%3A228&t=EkwGw0Pov8Q6NLVO-1

**Review by ahmdmhasn — CHANGES_REQUESTED** [review]

File: Healthy/Classes/Extensions/UIFont+Style.swift [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        }()
+
+        if let font = UIFont(name: fontName, size: size) {
+            return font
+        } else {
+            return UIFont.systemFont(ofSize: size, weight: weight == .regular ? .regular : .bold)
+        }
+
+    }
+}
```
Comment (ahmdmhasn) [inline Healthy/Classes/Extensions/UIFont+Style.swift]:
```suggestion
// MARK: add appFont func

extension UIFont {
    private enum PoppinsWeight {
        case regular
        case bold
    }

    private static func poppins(ofSize size: CGFloat, weight: PoppinsWeight) -> UIFont {
        let fontName: String = {
            switch weight {
            case .regular:
                return "Poppins-Regular"
            case .bold:
                return "Poppins-Bold"
            }
        }()

        guard let font = UIFont(name: fontName, size: size) else {
            assertionFailure("Unable to get a font with name: \(fontName)")
            return UIFont.systemFont(ofSize: size, weight: weight == .regular ? .regular : .bold)
        }
        
        return font
    }
}
```

Comment (ahmdmhasn) [reply Healthy/Classes/Extensions/UIFont+Style.swift]: I would suggest moving the getter of the font name to ba a computed property

```swift
        let fontName: String = {
            switch weight {
            case .regular:
                return "Poppins-Regular"
            case .bold:
                return "Poppins-Bold"
            }
        }()
```

File: Healthy/Classes/Extensions/UIFont+Style.swift [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+extension UIFont {
+    static let tittleTextRegular    = appFont(ofSize: 50, weight: .regular)
+    static let headerTextRegular    = appFont(ofSize: 30, weight: .regular)
+    static let largeTextRegular     = appFont(ofSize: 20, weight: .regular)
+    static let mediumTextRegular    = appFont(ofSize: 18, weight: .regular)
+    static let normalTextRegular    = appFont(ofSize: 16, weight: .regular)
+    static let smallTextRegular     = appFont(ofSize: 14, weight: .regular)
+    static let smallerTextRegular   = appFont(ofSize: 11, weight: .regular)
+}
```
Comment (ahmdmhasn) [inline Healthy/Classes/Extensions/UIFont+Style.swift]: Let's not store the fonts in memory, the system do this for us. Instead, please use computed properties.

Comment (ahmdmhasn) [reply Healthy/Classes/Extensions/UIFont+Style.swift]: Remove **"Text"** keyword, it's redundant.

**Comment by ahmdmhasn** [comment]
@abdallah-7698 This one as well

File: Healthy/Classes/Extensions/UIFont+Style.swift [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+extension UIFont {
+    static let tittleRegular    = appFont(ofSize: 50, weight: .regular)
+    static let headerRegular    = appFont(ofSize: 30, weight: .regular)
+    static let largeRegular     = appFont(ofSize: 20, weight: .regular)
+    static let mediumRegular    = appFont(ofSize: 18, weight: .regular)
+    static let normalRegular    = appFont(ofSize: 16, weight: .regular)
+    static let smallRegular     = appFont(ofSize: 14, weight: .regular)
+    static let smallerRegular   = appFont(ofSize: 11, weight: .regular)
+}
```
Comment (ahmdmhasn) [inline Healthy/Classes/Extensions/UIFont+Style.swift]:
```suggestion
import UIKit

// MARK: - Regular Fonts

extension UIFont {
    static var titleRegular: UIFont {
        poppins(ofSize: 50, weight: .regular)
    }
    
    static var headerRegular: UIFont {
        poppins(ofSize: 30, weight: .regular)
    }
    
    static var largeRegular: UIFont {
        poppins(ofSize: 20, weight: .regular)
    }
    
    static var mediumRegular: UIFont {
        poppins(ofSize: 18, weight: .regular)
    }
    
    static var normalRegular: UIFont {
        poppins(ofSize: 16, weight: .regular)
    }
    
    static var smallRegular: UIFont {
        poppins(ofSize: 14, weight: .regular)
    }
    
    static var smallerRegular: UIFont {
        poppins(ofSize: 11, weight: .regular)
    }
}

// MARK: - Bold Fonts

extension UIFont {
    static var titleBold: UIFont {
        poppins(ofSize: 50, weight: .bold)
    }
    
    static var headerBold: UIFont {
        poppins(ofSize: 30, weight: .bold)
    }
    
    static var largeBold: UIFont {
        poppins(ofSize: 20, weight: .bold)
    }
    
    static var mediumBold: UIFont {
        poppins(ofSize: 18, weight: .bold)
    }
    
    static var normalBold: UIFont {
        poppins(ofSize: 16, weight: .bold)
    }
    
    static var smallBold: UIFont {
        poppins(ofSize: 14, weight: .bold)
    }
    
    static var smallerBold: UIFont {
        poppins(ofSize: 11, weight: .bold)
    }
}

// MARK: Poppins UIFont Helpers

private extension UIFont {
    /// Returns a Poppins UIFont instance with the specified Style.
    static func poppins(ofSize size: CGFloat, weight: UIFont.Weight) -> UIFont {
        let fontName = poppinsFontName(forWeight: weight)
        guard let font = UIFont(name: fontName, size: size) else {
            assertionFailure("Unable to get a font with name: \(fontName)")
            return UIFont.systemFont(ofSize: size, weight: weight)
        }

        return font
    }
    
    /// Returns a Poppins UIFont file name for the given weight.
    static func poppinsFontName(forWeight weight: UIFont.Weight) -> String {
        switch weight {
        case .bold:
            return "Poppins-Bold"
        default:
            return "Poppins-Regular"
        }
    }
}
```

WDYT? 😁

**Review by ahmdmhasn — COMMENTED** [review]
Just a step! 🚀

**Review by oradwanomar — COMMENTED** [review]
Ahmed did a great review.
After resolve it's conversation tell me to approve
Great in any way Abdalla ❤️👌🏻

**Review by ahmdmhasn — APPROVED** [review]

**Pull request merged**